app/tests/spec/models/user.js

```diff
@@ -981,5 +981,94 @@ define(function (require, exports, module) {
         assert.isFalse(exists);
       });
     });
+
+    describe('beginFlow without flowId', function () {
+      beforeEach(function () {
+        sinon.stub(user, 'has', function () {
```

Comment:
You can simplify the housekeeping here using [sinon.sandbox.create](http://sinonjs.org/docs/#sandbox).

```js
var sandbox;
beforeEach(function () {
  sandbox = sinon.sandbox.create();
  sandbox.stub(user, 'has', ...
  sandbox.stub(user._flow, 'begin',...
  sandbox.stub(user._flow, 'get',...
  sandbox.spy(user, 'set');
  user.beginFlow('bar');
});

afterEach(function () {
  sandbox.restore();
});
```